Submit PBS jobs to specified queue

milljm opened the pull request:
added require_binaries param to allow unittesting of PBS functionality (qsub)

added a require_binaries unittest.
added a pbs unittest.

added option: --queue-queue
Use to specify which queue to submit jobs to (defaults to blank, which defaults to the default
queue).

I don't like the --queue-queue option name. But all the other options are prefixed with the word
'queue'. I feel we have no choice but to name it so.

Closes #12138

Commit:
Submit PBS jobs to specified queue

added require_binaries param to allow unittesting of PBS functionality (qsub).
added a require_binaries unittest.
added a pbs unittest to test basic qsub functionality.

added option: --queue-queue
Use to specify which queue to submit jobs to (defaults to blank, which defaults to the default
queue).

I don't like the --queue-queue option name. But all the other options are prefixed with the word
'queue'. I feel we have no choice but to name it so.

Closes #12138

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/232866/) on ae5fd15 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12149/site)

This comment will be updated on new commits.

Commit:
PBS feature update

Allow PBS to source a file before launching run_tests within the qsub script.
Create skippable PBS unittest based on availability.
Begin to deprecate --queue-meta options, in favor of --pbs-meta options.

Closes #12138

Commit:
Fix decorator

Handle exceptions caused by subprocess

Closes #12138

Commit:
fix missing requirements field

Closes #12138

Commit:
catch possible errors with --pbs-pre-source

Remove unused test spec file

Closes 12138

Commit:
Fix docstring typo

permcody commented on python/TestHarness/testers/Tester.py:
```diff
@@ -79,7 +79,7 @@ def validParams():
         params.addParam('check_input',    False, "Check for correct input file syntax")
         params.addParam('display_required', False, "The test requires and active display for rendering (i.e., ImageDiff tests).")
         params.addParam('timing',         True, "If True, the test will be allowed to run with the timing flag (i.e. Manually turning on performance logging).")
-        params.addParam('boost',         ['ALL'], "A test that runs only if BOOT is detected ('ALL', 'TRUE', 'FALSE')")
```
😆

permcody reviewed (approved):
Looks good!
